docs/weave-tutorial/05-messages.md

```diff
@@ -0,0 +1,114 @@
+---
+id: messages
+title: Messages
+sidebar_label: Messages
+---
+
+> [PR#2](https://github.com/iov-one/tutorial/pull/2): _Create msgs_
+
+Messages are requests for a change in the state, the action part of a transaction. They also need to be persisted (to be sent over the wire and stored on the blockchain), and must also be validated. They later are passed into [Handlers](https://godoc.org/github.com/iov-one/weave#Handler) to be processed and effect change in the blockchain state.
+
+## Messages vs. Transactions
+
+A message is a request to make change and this is the basic element of a blockchain. A transaction contains a message along with metadata and authorization information, such as fees, signatures, nonces, and time-to-live.
+
+A [Transaction](https://godoc.org/github.com/iov-one/weave#Tx) is fundamentally defined as anything persistent that holds a message:
+
+```go
+type Tx interface {
+    Persistent
+    // GetMsg returns the action we wish to communicate
+    GetMsg() (Msg, error)
+}
+```
+
+And every application can extend Tx with additional functionality, such as [Signatures](https://godoc.org/github.com/iov-one/weave/x/sigs#SignedTx), [Fees](https://godoc.org/github.com/iov-one/weave/x/cash#FeeTx), or anything else your application needs. The data placed in the Transaction is meant to be anything that applies to all modules, and is processed by a **Middleware**.
+
+A [Message](https://godoc.org/github.com/iov-one/weave#Msg) is also persistent and can be pretty much anything that an extension defines. The only necessary feature of a Message is `Path() string` method which provides the required route to the Handler.
+
+When we define a concrete transaction type for one application, we define it in protobuf with a set of possible messages that it can contain. Every application can add optional field to the transaction and allow a different set of messages, and the Handlers and Decorators work orthogonally to this, regardless of the **concrete** Transaction type.
```

Comment:
```suggestion
When we define a concrete transaction type for one application, we define it in protobuf with a set of possible messages that it can contain. Every application can add optional fields to the transaction and allow a different set of messages, and the Handlers and Decorators work orthogonally to this, regardless of the **concrete** Transaction type.
```